src/services/pcn-firewall/src/Chain.cpp

```diff
@@ -547,6 +535,11 @@ std::vector<std::shared_ptr<ChainRule>> Chain::getRuleList() {
 }
 
 void Chain::addRule(const uint32_t &id, const ChainRuleJsonObject &conf) {
+
+  if (id > rules_.size()) {
+    throw new std::runtime_error("rule id not allowed");
```

Comment:
I know this is fixed in a later commit, I think both commits should be squashed to this is fixed from the beginning.

Reply:
I squashed this commit.